cocos/core/components/component.ts

```diff
@@ -70,7 +70,13 @@ class Component extends CCObject {
         if (trimLeft >= 0) {
             className = className.slice(trimLeft + 1);
         }
-        return `${this.node.name}<${className}>`;
+
+        if (this.node) {
+            return `${this.node.name}<${className}>`;
+        } else {
+            error(`id = ${this.uuid}, className = ${className}, component's node is null or undefined.`);
```

Comment:
No need to report error here, this API shouldn't judge whether node should present or not. Just give the class name when node is absent